### ISSUE TITLE
Unable to Create Checkout using Line Items

### ISSUE THREAD

**Issue by zeeshan-plaza** [nozzlegear/ShopifySharp#797]
I have a verified sales channel app and even `write_checkout` permissions still I am unable to add email and line items in the checkout, the API call is successful and I get a valid checkout object but I am unable to attach line items and email

**Comment by nozzlegear** [comment]
Hey @zeeshan-plaza, could you post the code you're using to create a checkout? I've not personally used the sales channel APIs so I can't say what the issue would be off the top of my head.

**Comment by zeeshan-plaza** [comment]
> Hey @zeeshan-plaza, could you post the code you're using to create a checkout? I've not personally used the sales channel APIs so I can't say what the issue would be off the top of my head.
`
var lineItems = checkoutRequest.VariantIdQuantityMapping.Select(x => new ShopifySharp.CheckoutLineItem()
			{
				VariantId = Convert.ToInt64(variants[x.Key].ExternalId),
				Quantity = x.Value

			}).ToList();

			var checkoutService = new CheckoutService(storeUrl,shopifyAccessToken);

			var checkoutObj = new Checkout()
			{
				LineItems = lineItems,
				Email = "demo@gmail.com"
			};

			Checkout checkout = await checkoutService.CreateAsync(checkoutObj);

			Console.WriteLine(checkout);
`

I am attaching the ss of debugging as well, I am sending the line items and email in the correct way

<img width="516" alt="obj ss" src="https://user-images.githubusercontent.com/87425125/196282175-4d17c3f9-4fe4-471b-b421-627ea551bab8.png">

this is the response that i get email as well as line items are null but the checkout has been created , i get a checkout token as well
<img width="406" alt="line items" src="https://user-images.githubusercontent.com/87425125/196282302-2bb35510-74fd-441c-bfd6-e081d57778d6.png">

**Comment by nozzlegear** [comment]
Thanks! I've checked Shopify's docs and it looks like you're doing everything correctly, so my first thought is we've got some kind of serialization or deserialization problem happening in ShopifySharp. 

This is a little bit of extra work, but would you mind creating a temporary class to override the CheckoutService's `CreateAsync` method and then inspecting the `RawResult` string?

```
using System;
using System.Net.Http;
using System.Threading;
using System.Threading.Tasks;
using ShopifySharp.Infrastructure;

namespace ShopifySharp
{
    public class TempCheckoutService : CheckoutService
    {
        TempCheckoutService(string domain, string accessToken) : base(domain, accessToken) { }
        
        public override async Task<Checkout> CreateAsync(Checkout checkout, CancellationToken cancellationToken = default)
        {
            var req = PrepareRequest("checkouts.json");
            var body = checkout.ToDictionary();

            var result = await ExecuteRequestAsync(req, HttpMethod.Post, cancellationToken, new JsonContent(checkout));
            var inspectThis = result.RawResult;

            throw new NotImplementedException();
        }
    }
}
```

If you can post that value here that would help a ton. That will tell us if the line items are being created but ShopifySharp is not deserializing them properly, or if it's failing to send them at all.

**Comment by zeeshan-plaza** [comment]
> Thanks! I've checked Shopify's docs and it looks like you're doing everything correctly, so my first thought is we've got some kind of serialization or deserialization problem happening in ShopifySharp.
> 
> This is a little bit of extra work, but would you mind creating a temporary class to override the CheckoutService's `CreateAsync` method and then inspecting the `RawResult` string?
> 
> ```cs
> using System;
> using System.Net.Http;
> using System.Threading;
> using System.Threading.Tasks;
> using ShopifySharp.Infrastructure;
> 
> namespace ShopifySharp
> {
>     public class TempCheckoutService : CheckoutService
>     {
>         TempCheckoutService(string domain, string accessToken) : base(domain, accessToken) { }
>         
>         public override async Task<Checkout> CreateAsync(Checkout checkout, CancellationToken cancellationToken = default)
>         {
>             var req = PrepareRequest("checkouts.json");
>             var body = checkout.ToDictionary();
> 
>             var result = await ExecuteRequestAsync(req, HttpMethod.Post, cancellationToken, new JsonContent(checkout));
>             var inspectThis = result.RawResult;
> 
>             throw new NotImplementedException();
>         }
>     }
> }
> ```
> 
> If you can post that value here that would help a ton. That will tell us if the line items are being created but ShopifySharp is not deserializing them properly, or if it's failing to send them at all.

I tried this thing for debugging since the body variable was not used I commented it out and used overridden class

I got this in rawResult there are no line items :(  I'll attach an ss for the request at the bottom

```
{
  "checkout": {
    "completed_at": null,
    "created_at": "2022-10-18T03:32:34+05:30",
    "currency": "USD",
    "presentment_currency": "USD",
    "customer_id": null,
    "customer_locale": "en",
    "device_id": null,
    "discount_code": null,
    "discount_codes": [],
    "email": null,
    "legal_notice_url": null,
    "location_id": null,
    "name": "#23315457704098",
    "note": "",
    "note_attributes": {},
    "order_id": null,
    "order_status_url": null,
    "order": null,
    "payment_due": "0.00",
    "payment_url": "https://deposit.us.shopifycs.com/sessions",
    "payments": [],
    "phone": null,
    "shopify_payments_account_id": null,
    "privacy_policy_url": null,
    "refund_policy_url": null,
    "requires_shipping": false,
    "reservation_time_left": 0,
    "reservation_time": null,
    "source_identifier": null,
    "source_name": "12694847489",
    "source_url": null,
    "subscription_policy_url": null,
    "subtotal_price": "0.00",
    "shipping_policy_url": null,
    "tax_exempt": false,
    "taxes_included": false,
    "terms_of_sale_url": null,
    "terms_of_service_url": null,
    "token": "ab4f423df2553756fe3745c4767a77ef",
    "total_price": "0.00",
    "total_tax": "0.00",
    "total_tip_received": "0.00",
    "total_line_items_price": "0.00",
    "updated_at": "2022-10-18T03:32:34+05:30",
    "user_id": null,
    "web_url": "https://my-store.myshopify.com/62060265634/checkouts/ab4f423df2553756fe3795c4767a77ef",
    "line_items": [],
    "gift_cards": [],
    "tax_lines": [],
    "tax_manipulations": [],
    "shipping_line": null,
    "shipping_rate": null,
    "shipping_address": null,
    "credit_card": null,
    "billing_address": null,
    "applied_discount": null,
    "applied_discounts": [],
    "discount_violations": []
  }
}
```
<img width="690" alt="Dummy Class stuff" src="https://user-images.githubusercontent.com/87425125/196292973-c94bece8-5f39-4524-9aa7-09d1aa426e37.png">

**Comment by zeeshan-plaza** [comment]
Hey @nozzlegear  For the above code snippet if i replace the

`
 var result = await ExecuteRequestAsync(req, HttpMethod.Post, cancellationToken, new JsonContent(checkout));`

with 

`
 var result = await ExecuteRequestAsync(req, HttpMethod.Post, cancellationToken, new JsonContent(new {checkout}));`

then it works like a charm so is it like I have to send it in a new object ??, I tried this because there is a checkout wrapper in the rest API body.

This is what I got when I tried sending the JsonContent in the new object this contains email and line items , the perfect response :)

```
{
  "checkout": {
    "completed_at": null,
    "created_at": "2022-10-18T03:50:38+05:30",
    "currency": "USD",
    "presentment_currency": "USD",
    "customer_id": 6548725366946,
    "customer_locale": "en",
    "device_id": null,
    "discount_code": null,
    "discount_codes": [],
    "email": "demo@gmail.com",
    "legal_notice_url": null,
    "location_id": null,
    "name": "#23315542999202",
    "note": "",
    "note_attributes": {},
    "order_id": null,
    "order_status_url": null,
    "order": null,
    "payment_due": "0.00",
    "payment_url": "https://deposit.us.shopifycs.com/sessions",
    "payments": [],
    "phone": null,
    "shopify_payments_account_id": null,
    "privacy_policy_url": null,
    "refund_policy_url": null,
    "requires_shipping": true,
    "reservation_time_left": 0,
    "reservation_time": null,
    "source_identifier": null,
    "source_name": "12694847489",
    "source_url": null,
    "subscription_policy_url": null,
    "subtotal_price": "0.00",
    "shipping_policy_url": null,
    "tax_exempt": false,
    "taxes_included": false,
    "terms_of_sale_url": null,
    "terms_of_service_url": null,
    "token": "b2382bf1879883133127a39f73e3ed26",
    "total_price": "0.00",
    "total_tax": "0.00",
    "total_tip_received": "0.00",
    "total_line_items_price": "0.00",
    "updated_at": "2022-10-18T03:50:38+05:30",
    "user_id": null,
    "web_url": "https://store-demo-store.myshopify.com/62060265634/checkouts/b2382bf1879883173127a39f73e3ed26",
    "line_items": [
      {
        "id": "d7747d6bfe087ce51f504fb176cf484a",
        "key": "d7748d6bfe087ce51f504fb176cf484a",
        "product_id": 7700781269154,
        "variant_id": 42732838453410,
        "sku": "5",
        "vendor": "Checkout Demo Store",
        "title": "Tshirt with zero price",
        "variant_title": "",
        "image_url": "//cdn.shopify.com/shopifycloud/shopify/assets/checkout/product-blank-98d4187c2152136e9fb0587a99dfcce6f6873f3a9f21ea9135ed7f495296090f.png",
        "taxable": false,
        "requires_shipping": true,
        "gift_card": false,
        "price": "0.00",
        "compare_at_price": "0.00",
        "line_price": "0.00",
        "properties": {},
        "quantity": 1,
        "grams": 0,
        "fulfillment_service": "manual",
        "applied_discounts": [],
        "discount_allocations": [],
        "tax_lines": []
      }
    ],
    "gift_cards": [],
    "tax_lines": [],
    "tax_manipulations": [],
    "shipping_line": null,
    "shipping_rate": null,
    "shipping_address": null,
    "credit_card": null,
    "billing_address": null,
    "applied_discount": null,
    "applied_discounts": [],
    "discount_violations": []
  }
}
```

**Comment by nozzlegear** [comment]
Aha awesome catch! That's exactly what it should be doing, and that's what most of our methods in Shopifysharp do as well. Looks like you've found and fixed a bug. If you want to submit a pull request for the CheckoutService I'd be happy to merge and publish it, otherwise I'll get it patched by tomorrow!

**Comment by nozzlegear** [comment]
@zeeshan-plaza The fix has now been published on nuget in version 5.18.6. Thanks for helping debug and fix this!
